paddle/fluid/operators/detail/grpc_server.h

```diff
@@ -53,6 +55,12 @@ class AsyncGRPCServer final {
 
   void SetDevCtx(const platform::DeviceContext *dev_ctx) { dev_ctx_ = dev_ctx; }
 
+  void SetProgram(framework::ProgramDesc *program) { program_ = program; }
+
+  void SetPrefetchBlkdId(int blkid) { prefetch_blk_id_ = blkid; }
+
+  void SetExecutor(framework::Executor *executor) { executor_ = executor; }
```

Comment:
Is our executor.Run thread safe?

Reply:
Not yet, on PServer, all variables are in the same scope without any condition control, but we can use different blocks to run Op parallel.For now, we would use `Distribute transpiler` to make sure there are no any two Ops sharing one same variable.